pkg/btf/btf.go

```diff
@@ -329,3 +355,30 @@ func processMembers(
 	}
 	return &currentType, nil
 }
+
+func processArray(
+	btfArgs *[api.MaxBTFArgDepth]api.ConfigBTFArg,
+	currentType btf.Type,
+	pathToFound []string,
+	i int,
+	idx uint32,
+) (*btf.Type, error) {
+	var target btf.Type
+	if t, ok := currentType.(*btf.Array); ok {
+		target = ResolveNestedTypes(t.Type)
+	} else if t, ok := currentType.(*btf.Pointer); ok {
+		target = ResolveNestedTypes(t.Target)
+		(*btfArgs)[i].IsPointer = uint16(1)
+		if i > 0 {
+			(*btfArgs)[i-1].IsPointer = uint16(1)
+		}
```

Comment:
so all the work done with the pointer to you fixed size array looks a bit hard to read, I don't have much time to propose something now but I wonder if you could simplify this code.

Do you really need a pointer to your array? It's already pointing to your array elements.